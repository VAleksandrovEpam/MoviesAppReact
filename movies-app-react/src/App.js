import './App.css';
import HomeHeader from './components/homepage/components/header/HeaderHome';

function App() {
  return (
    <>
      <HomeHeader></HomeHeader>
    </>
  );
}

export default App;
