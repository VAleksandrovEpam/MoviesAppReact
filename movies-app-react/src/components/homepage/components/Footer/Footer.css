.primary_color {
    color: #F65261;
    margin: 0;
}

.primary_color > span {
    font-weight: 800;
}

footer {
    width: 80%;
    margin: auto;
    background-color: #424242;
    text-align: center;
    padding: 1rem 0rem;
}