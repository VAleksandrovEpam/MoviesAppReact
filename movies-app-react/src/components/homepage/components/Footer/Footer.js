import './Footer.css';
import { useLocation, useHistory } from "react-router";
import { useEffect } from "react";

function Footer() {
    const location = useLocation();
    const history = useHistory();
    console.log(location.pathname)
    useEffect(() => {
        if (location.pathname === "/") {
            history.push("search");
          }
    }, [location.pathname])
    return (
        <>
            <footer>
                <p className="primary_color"><span>netflix</span>roulette</p>
            </footer>
        </>
    )
}

export default Footer;