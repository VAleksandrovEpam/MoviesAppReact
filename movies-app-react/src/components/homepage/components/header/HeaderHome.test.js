import renderer from 'react-test-renderer';
window.React = require('react')
import App from '../../../../App';
import HomeHeader from './HeaderHome';
import Footer from '../Footer/Footer';
import { BrowserRouter as Router } from "react-router-dom";

fdescribe("HomeHeader Component Testing", () => {
  test("snapshot testing" , () => {
    const homHeader = renderer
    .create(
      <App>
        <Router>
          <HomeHeader></HomeHeader>
          <Footer></Footer>
        </Router>      
      </App>
   ).toJSON();
  expect(homHeader).toMatchSnapshot()
  });
})