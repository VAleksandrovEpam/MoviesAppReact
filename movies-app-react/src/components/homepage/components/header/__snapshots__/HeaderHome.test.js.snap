// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<footer>
  <p
    className="primary_color"
  >
    <span>
      netflix
    </span>
    roulette
  </p>
</footer>
`;

exports[`HomeHeader Component Testing snapshot testing 1`] = `
<footer>
  <p
    className="primary_color"
  >
    <span>
      netflix
    </span>
    roulette
  </p>
</footer>
`;
